### ISSUE TITLE
Programatically force/disable the colors

### ISSUE THREAD

**Issue by mackwic** [mackwic/colored#4]
It would be great to have a static API which would enable/disable coloring. It would help for a finer control.

Example 1:

``` rust
fn log_something(msg: String) {
    colored::force_colors(Some(ApplyColors::Never));  // never color
    log_with_colors_in_file(msg);                     // no colors applied
    colored::force_colors(Some(ApplyColors::Always)); // always color when asked
    log_with_colors_in_stdout(msg);                   // colors applied
    colored::force_colors(None);
    // back to default behavior
}
```

It could be with a color token (Example 2):

``` rust
fn log_something(msg: String) {
    let token = colored::force_colors_scoped(Some(ApplyColors::Never));  // never color
    log_with_colors_in_file(msg);                     // no colors applied
    // first token goes out of scope, restoring default behavior
    // second token is emited
    let token = colored::force_colors_scoped(Some(ApplyColors::Always)); // always color when asked
    log_with_colors_in_stdout(msg);                   // colors applied
    // back to default behavior because second token goes out of scope
}
```

**Comment by mackwic** [comment]
fixed by #10

**Comment by Fitzsimmons** [comment]
Can you give an example of how this is used? I can't figure it out.

**Comment by mackwic** [comment]
Sure thing !

Let's say we have this program:

```rust
// main.rs
extern crate colored;
use colored::*;

fn main() {
     println!("{}", "some warning".yellow());
}
```

If you `cargo run` it, you'll get some warning in yellow. But, if you use environment like that: `CLI_COLOR=0 cargo run`, there will be no color.

This is control by environment.

You can also control by function call like that:

```rust
// main.rs
extern crate colored;
use colored::*;

fn main() {
    // this will be yellow if your environment allow it
    println!("{}", "some warning".yellow());
    // now , this will be always yellow
    colored::control::SHOULD_COLORIZE.set_override(true);
    println!("{}", "some warning".yellow());
    // now, this will be never yellow
    colored::control::SHOULD_COLORIZE.set_override(true);
    println!("{}", "some warning".yellow());
    // let the environment decide again
    colored::control::SHOULD_COLORIZE.unset_override();
}
```
... Aaaaand, this should be how it works **but it doesn't** because I obviously forgot to add an example and so I didn't caught the `cannot borrow immutable borrowed content as mutable` error.

My bad. I am really sorry, this shouldn't had happened

I'll fix it tomorrow, with an example and an helper function in `colored:control`.

**Comment by mackwic** [comment]
Fixed in #16 with new example and helper as promised 👍

**Comment by mackwic** [comment]
Published in v1.4.1
